paddle/fluid/framework/ddim.h

```diff
@@ -126,6 +126,67 @@ class DDim {
 
   std::string to_str() const;
 
+  DDim reshape(const std::vector<int>& shape) const {
+    const int64_t copy_dim_val = 0;
+    const DDim& in_dims = *this;
+    DDim out_dims;
+    out_dims.rank_ = shape.size();
+    for (size_t i = 0; i < shape.size(); ++i) {
+      if (shape[i] == copy_dim_val) {
+        PADDLE_ENFORCE_LT(static_cast<int>(i), in_dims.size(),
+                          platform::errors::InvalidArgument(
+                              "The index of 0 in `shape` must be less than "
+                              "the dimensions. But received shape[%d] "
```

Comment:
`The index of 0 in `shape` must be less than the dimensions.`

Could you please clarify the message? @sfraczek Could you help here?
My best guess is that "Index %d of shape under which the value of 0 is stored, must be lower than the **number** of old dimensions." was meant.

Reply:
Sounds good to me. We can take your suggestion.

Reply:
@Sand3r-  Should we keep the part "But received shape[...."?

Reply:
Oh yes, totally :)

Reply:
done